build/source/driver/multi_driver.f90

```diff
@@ -463,28 +500,26 @@ program multi_driver
 ! *****************************************************************************
 ! (5c) read trial model parameter values for each HRU, and populate initial data structures
 ! *****************************************************************************
-call read_param(nHRU,typeStruct,mparStruct,err,message); call handle_err(err,message)
+call read_param(nGRU,typeStruct,mparStruct,err,message); call handle_err(err,message)
 
 ! *****************************************************************************
 ! (5d) compute derived model variables that are pretty much constant for the basin as a whole
 ! *****************************************************************************
 
 ! loop through GRUs
 do iGRU=1,nGRU
-
  ! calculate the fraction of runoff in future time steps
  call fracFuture(bparStruct%gru(iGRU)%var,    &  ! vector of basin-average model parameters
                  bvarStruct%gru(iGRU),        &  ! data structure of basin-average variables
                  err,message)                    ! error control
  call handle_err(err,message)
 
  ! loop through local HRUs
- do iHRU=1,gru_struc(iGRU)%hruCount
-
+ do localHRU=1,gru_struc(iGRU)%hruCount
   kHRU=0
-  ! check the network topology (only expect there to be one downslope HRU)
-  do jHRU=1,nHRU
-   if(typeStruct%gru(iGRU)%hru(iHRU)%var(iLookTYPE%downHRUindex) == typeStruct%gru(iGRU)%hru(jHRU)%var(iLookTYPE%hruIndex))then
+  ! check the network topology (only expect there to be one downslope HRU) ! multiple downslope HRUs only occur if there are multiple var(iLookTYPE%downHRUindex). this section looks like checking duplicate hruId (multiple hruIndex == current downHRUindex)?
+  do jHRU=1,gru_struc(iGRU)%hruCount
+   if(typeStruct%gru(iGRU)%hru(localHRU)%var(iLookTYPE%downHRUindex) == typeStruct%gru(iGRU)%hru(jHRU)%var(iLookTYPE%hruIndex))then
```

Comment:
I am a bit confused with this block. 

Is it testing one HRU has one and only one downstream HRU?